### PR TITLE
Remove  unused code to save gas costs in Cluster Rewards contract

### PR DESCRIPTION
**Pull request by akshay111meher**
Remove feed

### CODE REVIEW

File: contracts/staking/ClusterRewards.sol [hunk 1]
```diff
@@ -464,13 +432,8 @@ contract ClusterRewards is
...
 
-    function _getRewardForEpoch(uint256 _epoch, bytes32 _networkId, uint256 _epochLength) internal view returns (uint256) {
-        if (_epoch < SWITCHING_PERIOD / _epochLength) return 0;
-        return (totalRewardsPerEpoch * rewardWeight[_networkId]) / totalRewardWeight;
-    }
-
-    function getRewardForEpoch(uint256 _epoch, bytes32 _networkId) public view returns (uint256) {
-        return _getRewardForEpoch(_epoch, _networkId, receiverStaking.EPOCH_LENGTH());
+    function getRewardForEpoch(bytes32 _networkId) public view returns (uint256) {
+        return (totalRewardsPerEpoch * rewardWeight[_networkId]) / receiverStaking.EPOCH_LENGTH();
```
Comment (prateekreddy) [inline contracts/staking/ClusterRewards.sol]: This is incorrect ?

File: contracts/staking/ClusterRewards.sol [hunk 2]
```diff
@@ -346,12 +315,11 @@ contract ClusterRewards is
         address[][] memory _selectedClusters = clusterSelectors[_networkId].getClustersRanged(_fromEpoch, _noOfEpochs);
 
         ReceiverPayment memory receiverPayment = receiverRewardPayment[_receiver];
-        uint256 _totalNetworkRewardsPerEpoch;
+        // _totalNetworkRewardsPerEpoch = inflation rewards + receiver rewards
```
Comment (prateekreddy) [inline contracts/staking/ClusterRewards.sol]: @akshay111meher This comment doesn't seem right, can you recheck and place is appropriately.

File: contracts/staking/ClusterRewards.sol [hunk 3]
```diff
@@ -464,13 +432,8 @@ contract ClusterRewards is
         return 0;
     }
 
-    function _getRewardForEpoch(uint256 _epoch, bytes32 _networkId, uint256 _epochLength) internal view returns (uint256) {
-        if (_epoch < SWITCHING_PERIOD / _epochLength) return 0;
```
Comment (prateekreddy) [inline contracts/staking/ClusterRewards.sol:468]: Constant `SWITCHING_PERIOD` can be removed as it is not used anywhere

File: contracts/staking/ClusterRewards.sol [hunk 4]
```diff
@@ -220,32 +215,6 @@ contract ClusterRewards is
 
     //-------------------------------- Admin functions end --------------------------------//
 
-    //-------------------------------- User functions start --------------------------------//
-
-    function feed(bytes32 _networkId, address[] calldata _clusters, uint256[] calldata _payouts, uint256 _epoch) external onlyFeeder {
-        require(receiverStaking.START_TIME() + SWITCHING_PERIOD + 1 days > block.timestamp, "CRW:F-Invalid method");
-        uint256 rewardDistributed = rewardDistributedPerEpoch[_epoch];
```
Comment (prateekreddy) [inline contracts/staking/ClusterRewards.sol:227]: Rename all variables which aren't used anymore with `unused` like  we do  in other contracts.

**Review by prateekreddy — CHANGES_REQUESTED** [review]

File: contracts/staking/ClusterRewards.sol (hunk 2 above)
Comment (akshay111meher) [reply contracts/staking/ClusterRewards.sol]: removed

File: contracts/staking/ClusterRewards.sol (hunk 3 above)
Comment (akshay111meher) [reply contracts/staking/ClusterRewards.sol:468]: removed

File: contracts/staking/ClusterRewards.sol (hunk 4 above)
Comment (akshay111meher) [reply contracts/staking/ClusterRewards.sol:227]: removed

File: contracts/staking/ClusterRewards.sol (hunk 1 above)
Comment (akshay111meher) [reply contracts/staking/ClusterRewards.sol]: Changed

File: contracts/staking/ClusterRewards.sol [hunk 5]
```diff
@@ -109,31 +109,28 @@ contract ClusterRewards is
...
-    mapping(address => uint256) public clusterRewards;
+    mapping(address => uint256) public override clusterRewards;
 
-    mapping(bytes32 => uint256) public rewardWeight;
+    mapping(bytes32 => uint256) public override rewardWeight;
     uint256 public totalRewardWeight;
-    uint256 public totalRewardsPerEpoch;
-    uint256 public payoutDenomination;
+    uint256 public override totalRewardsPerEpoch;
+    uint256 public unsed_payoutDenomination;
```
Comment (prateekreddy) [inline contracts/staking/ClusterRewards.sol]: typo

File: test/staking/ClusterRewards.ts [hunk 6]
```diff
@@ -1290,23 +1072,23 @@ describe("ClusterRewards submit compressed tickets", function () {
       expect(await clusterRewards.isTicketsIssued(addrs[5], startEpoch + i)).to.be.false;
       expect(await clusterRewards.clusterRewards(addrs[31])).to.be.closeTo(
         receiverRewards1[0].add(receiverRewards2[4]).add(extraRewards1[0]).add(extraRewards2[4]),
-        11
+        20
```
Comment (prateekreddy) [inline test/staking/ClusterRewards.ts]: What is the % error here ?

Comment (akshay111meher) [reply test/staking/ClusterRewards.ts]: changed to percentageDelta.div(e18)

**Review by prateekreddy — APPROVED** [review]

**Pull request merged**